cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/etl/batch/sink/DBSinkTest.java

```diff
@@ -0,0 +1,162 @@
+    Assert.assertEquals(
+      getRecordFields(input),
+      getRecordFields(emitter.getEmitted().get(0).getKey().getRecord()));
+  }
+
+  @Test
+  public void testDropColumn() throws Exception {
+    final DBSink.DBSinkConfig config = new DBSink.DBSinkConfig();
+    config.columns = "body,ts";
+    config.tableName = "foo";
+
+    DBSink sink = new DBSink(config) {
+      @Override
+      public List<String> getColumns() {
+        return ImmutableList.copyOf(Splitter.on(",").split(config.columns));
+      }
+    };
+
+    StructuredRecord input = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
+        Schema.Field.of("headers", Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING))),
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING))))
+      .set("ts", 123)
+      .set("headers", ImmutableMap.of("hello", "zzz"))
+      .set("body", "sdfsdf")
+      .build();
+
+    StructuredRecord output = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING)),
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG))))
+      .set("body", "sdfsdf")
+      .set("ts", 123)
+      .build();
+
+    MockEmitter<KeyValue<DBRecord, NullWritable>> emitter = new MockEmitter<>();
+    sink.transform(input, emitter);
+
+    Assert.assertEquals(1, emitter.getEmitted().size());
+    Assert.assertEquals(
+      getRecordFields(output),
+      getRecordFields(emitter.getEmitted().get(0).getKey().getRecord()));
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testFailure() throws Exception {
+    final DBSink.DBSinkConfig config = new DBSink.DBSinkConfig();
+    config.columns = "body,ts,missing";
+    config.tableName = "foo";
+
+    DBSink sink = new DBSink(config) {
+      @Override
+      public List<String> getColumns() {
+        return ImmutableList.copyOf(Splitter.on(",").split(config.columns));
+      }
+    };
+
+    StructuredRecord input = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG)),
+        Schema.Field.of("headers", Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING))),
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING))))
+      .set("ts", 123)
+      .set("headers", ImmutableMap.of("hello", "zzz"))
+      .set("body", "sdfsdf")
+      .build();
+
+    StructuredRecord output = StructuredRecord
+      .builder(Schema.recordOf(
+        "foo",
+        Schema.Field.of("body", Schema.of(Schema.Type.STRING)),
+        Schema.Field.of("ts", Schema.of(Schema.Type.LONG))))
+      .set("body", "sdfsdf")
+      .set("ts", 123)
+      .build();
+
+    MockEmitter<KeyValue<DBRecord, NullWritable>> emitter = new MockEmitter<>();
+    sink.transform(input, emitter);
+
+    Assert.assertEquals(1, emitter.getEmitted().size());
```

Comment:
are these assertions needed if there is an exception? there shouldn't be anything in the emitter in this case, right?